Source/ACE/Entity/Creature.cs

```diff
@@ -251,115 +251,227 @@ protected virtual void VitalTickInternal(CreatureVital vital)
-                            {
-                                gm = new UniversalMotion(MotionStance.MeleeShieldAttack);
-                                gm.MovementData.CurrentStyle = (ushort)MotionStance.MeleeShieldAttack;
-                            }
-                            else
-                            {
-                                gm = new UniversalMotion(MotionStance.ThrownShieldCombat);
-                                gm.MovementData.CurrentStyle = (ushort)MotionStance.ThrownShieldCombat;
-                            }
-                        }
-                    }
-                    // If I am down to here and still null - I have no idea - do nothing.
-                    if (gm != null)
-                        SetMotionState(gm);
+                WorldObject missileWeapon = GetInventoryItem(new ObjectGuid(mEquipedMissile.Guid));
+                if (missileWeapon == null)
+                {
+                    log.InfoFormat("Changing combat mode for {0} - could not locate wielded weapon {1}", Guid, mEquipedMissile.Guid);
+                    return;
+                }
+                EquippedItem mEquipedAmmo = Children.Find(s => s.EquipMask == EquipMask.MissileAmmo);
+                MotionStance ms;
+
+                if (missileWeapon.DefaultCombatStyle != null)
+                    ms = (MotionStance)missileWeapon.DefaultCombatStyle;
+                else
+                {
+                    log.InfoFormat("Changing combat mode for {0} - wielded item {1} has not be assigned a default combat style", Guid, mEquipedMissile.Guid);
+                    return;
+                }
+
+                UniversalMotion mm = new UniversalMotion(ms);
+                mm.MovementData.CurrentStyle = (ushort)ms;
+                if (mEquipedAmmo == null)
+                {
+                    CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+                    SetMotionState(this, mm);
+                }
+                else
+                {
+                    WorldObject ammo = GetInventoryItem(new ObjectGuid(mEquipedAmmo.Guid));
+                    CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+                    SetMotionState(this, mm);
+                    mm.MovementData.ForwardCommand = (ushort)MotionCommand.Reload;
+                    SetMotionState(this, mm);
+                    CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange, new GameMessageUpdatePosition(this));
+                    combatModeChain.AddDelaySeconds(0.25);
+                    // System.Threading.Thread.Sleep(250); // used for debugging
+                    mm.MovementData.ForwardCommand = (ushort)MotionCommand.Invalid;
+                    SetMotionState(this, mm);
+                    combatModeChain.AddDelaySeconds(0.40);
```

Comment:
I put in FIXME's on both of these until we can get the delay times.   Without these delays, it does not function correctly.   However, just FYI these are about the correct delays - it will be interesting to see what the actual delays are.